ApplicationLibCode/ProjectDataModel/Faults/RimFaultReactivationDataAccessorGeoMech.cpp

```diff
@@ -0,0 +1,162 @@
+///
+//--------------------------------------------------------------------------------------------------
+RimFaultReactivationDataAccessorGeoMech::RimFaultReactivationDataAccessorGeoMech( RimGeoMechCase*                geoMechCase,
+                                                                                  RimFaultReactivation::Property property )
+    : m_geoMechCase( geoMechCase )
+    , m_property( property )
+{
+    m_geoMechCaseData = geoMechCase->geoMechData();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimFaultReactivationDataAccessorGeoMech::~RimFaultReactivationDataAccessorGeoMech()
+{
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimFaultReactivationDataAccessorGeoMech::updateResultAccessor()
+{
+    const int partIndex = 0;
+
+    auto loadFrameLambda = [&]( auto femParts, RigFemResultAddress addr ) -> RigFemScalarResultFrames*
+    {
+        auto result = femParts->findOrLoadScalarResult( partIndex, addr );
+        if ( result->frameData( 0, 0 ).empty() )
+        {
+            return nullptr;
+        }
+        return result;
+    };
+
+    auto femParts  = m_geoMechCaseData->femPartResults();
+    m_resultFrames = loadFrameLambda( femParts, getResultAddress( m_property ) );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RigFemResultAddress RimFaultReactivationDataAccessorGeoMech::getResultAddress( RimFaultReactivation::Property property )
+{
+    if ( property == RimFaultReactivation::Property::YoungsModulus ) return RigFemResultAddress( RIG_ELEMENT, "MODULUS", "" );
+    if ( property == RimFaultReactivation::Property::PoissonsRatio ) return RigFemResultAddress( RIG_ELEMENT, "RATIO", "" );
+    CAF_ASSERT( property == RimFaultReactivation::Property::Density );
+    return RigFemResultAddress( RIG_ELEMENT, "DENSITY", "" );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RimFaultReactivationDataAccessorGeoMech::isMatching( RimFaultReactivation::Property property ) const
+{
+    return property == m_property;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+double RimFaultReactivationDataAccessorGeoMech::valueAtPosition( const cvf::Vec3d& position ) const
+{
```

Comment:
Could this be done with less or shared code? Ref. RimWellIADataAccess.cpp

Reply:
Good idea! Fixed.